lib/core/src/display/navigation/mouse_manager.rs

```diff
@@ -0,0 +1,295 @@
+
+/// An enumeration representing the mouse buttons.
+pub enum MouseButton {
+    LEFT,
+    MIDDLE,
+    RIGHT
+}
+
+// =======================
+// === MouseClickEvent ===
+// =======================
+
+/// A struct storing information about mouse down and mouse up events.
+pub struct MouseClickEvent {
+    pub position : Vector2<f32>,
+    pub button   : MouseButton
+}
+
+impl MouseClickEvent {
+    fn from(event:MouseEvent, data:&Rc<RefCell<MouseManagerData>>) -> Self {
+        let position  = Vector2::new(event.x() as f32, event.y() as f32);
+        let position  = position - data.borrow().dom.position();
+        let button    = match event.button() {
+            LEFT_MOUSE_BUTTON      => MouseButton::LEFT,
+            MIDDLE_MOUSE_BUTTON    => MouseButton::MIDDLE,
+            RIGHT_MOUSE_BUTTON | _ => MouseButton::RIGHT
+        };
+        Self { position, button }
+    }
+}
+
+pub trait FnMouseClick = Fn(MouseClickEvent) + 'static;
+
+// ==========================
+// === MousePositionevent ===
+// ==========================
+
+/// A struct storing information about mouse move, mouse enter and mouse leave events.
+pub struct MousePositionEvent {
+    pub previous_position : Vector2<f32>,
+    pub position          : Vector2<f32>
+}
+
+impl MousePositionEvent {
+    fn from(event:MouseEvent, data:&Rc<RefCell<MouseManagerData>>) -> Self {
+        let position  = Vector2::new(event.x() as f32, event.y() as f32);
+        let position  = position - data.borrow().dom.position();
+        let previous_position = match data.borrow().mouse_position {
+            Some(position) => position,
+            None           => position
+        };
+        data.borrow_mut().mouse_position = Some(position);
+        Self { previous_position, position }
+    }
+}
+
+pub trait FnMousePosition = Fn(MousePositionEvent)  + 'static;
+
+// =======================
+// === MouseWheelEvent ===
+// =======================
+
+/// A struct storing information about mouse wheel events.
+pub struct MouseWheelEvent {
+    pub movement_y : f32
+}
+
+impl MouseWheelEvent {
+    fn from(event:WheelEvent) -> Self {
+        let movement_y = event.delta_y() as f32;
+        Self { movement_y }
+    }
+}
+
+pub trait FnMouseWheel = Fn(MouseWheelEvent) + 'static;
+
+// ========================
+// === MouseManagerData ===
+// ========================
+
+/// A struct used for storing shared MouseManager's mutable data.
+struct MouseManagerData {
+    dom            : DOMContainer,
+    mouse_position : Option<Vector2<f32>>
+}
+
+// =============
+// === State ===
+// =============
+
+/// An enum mainly used for enabling or disabling the Context Menu.
+pub enum State {
+    Enabled,
+    Disabled
+}
+
+// ====================
+// === MouseManager ===
+// ====================
+
+/// MouseManager
+pub struct MouseManager {
+    target : EventTarget,
+    data   : Rc<RefCell<MouseManagerData>>,
+    ignore_context_menu : Option<Closure<dyn Fn(MouseEvent)>>
```

Comment:
Why `ignore_context_menu` is an optional function? It sounds like it was a bool. Either the name is bad or lack of docs here.